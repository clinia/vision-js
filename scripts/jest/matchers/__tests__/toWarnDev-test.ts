/* eslint-disable no-console */

describe('toWarnDev', () => {
  describe('usage', () => {
    test('fails with incorrect type of message', () => {
      expect(() => {
        // @ts-ignore:next-line
        expect(() => {}).toWarnDev(false);
      }).toThrowErrorMatchingSnapshot();
    });
  });

  if (__DEV__) {
    describe('without message', () => {
      test('does not fail if called', () => {
        expect(() => {
          expect(() => {
            console.warn('warning');
          }).toWarnDev();
        }).not.toThrow();
      });

      test('fails if not called', () => {
        expect(() => {
          expect(() => {}).toWarnDev();
        }).toThrowErrorMatchingSnapshot();
      });
    });

    describe('with message', () => {
      test('does not fail with correct message', () => {
        expect(() => {
          expect(() => {
            console.warn('warning');
          }).toWarnDev('warning');
        }).not.toThrow();
      });

      test('fails if a warning is not correct', () => {
        expect(() => {
          expect(() => {
            console.warn('warning');
          }).toWarnDev('another warning');
        }).toThrow(/Unexpected warning recorded./);
      });
    });
  }
});

/* eslint-enable no-console */
