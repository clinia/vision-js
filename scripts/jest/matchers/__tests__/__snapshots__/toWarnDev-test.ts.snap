// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toWarnDev usage fails with incorrect type of message 1`] = `"toWarnDev() requires a parameter of type string but was given boolean."`;

exports[`toWarnDev without message fails if not called 1`] = `"No warning recorded."`;
